c7n/resources/glue.py

```diff
@@ -208,3 +208,51 @@ def process(self, resources):
                 client.delete_job(JobName=r['Name'])
             except client.exceptions.EntityNotFoundException:
                 raise
+
+
+@resources.register('glue-crawler')
+class GlueCrawler(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'glue'
+        enum_spec = ('get_crawlers', 'Crawlers', None)
+        detail_spec = None
+        id = name = 'Name'
+        date = 'CreatedOn'
+        dimension = None
+        filter_name = None
+        arn = False
+
+    permissions = ('glue:GetCrawlers',)
+
+    @property
+    def generate_arn(self):
+        self._generate_arn = functools.partial(
+            generate_arn,
+            'glue',
+            region=self.config.region,
+            account_id=self.config.account_id,
+            resource_type='crawler',
+            separator='/')
+        return self._generate_arn
+
+    def get_arns(self, resources):
+        return [self.generate_arn(r['Name']) for r in resources]
+
+
+register_universal_tags(GlueCrawler.filter_registry, GlueCrawler.action_registry)
+
+
+@GlueCrawler.action_registry.register('delete')
+class DeleteCrawler(BaseAction):
+
+    schema = type_schema('delete')
+    permissions = ('glue:DeleteCrawler',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('glue')
+        for r in resources:
+            try:
+                client.delete_crawler(Name=r['Name'])
+            except client.exceptions.EntityNotFoundException:
+                raise
```

Comment:
oops i left one of these through on the glue jobs pr, please fix that in this pr as well (for delete action). the catch should be continue.

Reply:
also of note, delete is only available on crawlers not in the RUNNING state https://docs.aws.amazon.com/glue/latest/webapi/API_DeleteCrawler.html

Reply:
will do thanks

Reply:
per thisishi's comments this will also need a state filter for those in the running state

Reply:
It seems delete also does not work for crawlers in STOPPING state as well as RUNNING.

Reply:
Would it be worth it to make the Ec2.StateTransitionFilter more generic here? other actions from other resources have this issue as well, maybe add a `state` jmespath expression on the resource definition to support this

then this changes to:

c7n.resources.glue:
```python
valid_origin_states = ('READY',)

def process(self, resources):
  resources = self.filter_resource_state(resources)
  try:
    ...
```

c7n.resources.ec2:
```python
class StateTransitionFilter(object):
  valid_origin_states = ()
  
  def filter_resource_state(self, resources):
    filtered = []
    state = self.manager.get_model().state
    state_resource_map = zip(jmespath.compile('[].%s' % state, resources), resources)
    for k, v in state_resource_map:
      if k in self.valid_origin_states:
        filtered.append(v)
    return filtered
```

Reply:
oh nice I can add StateTransitionFilter for glue resources

Reply:
Let me know if this is what you were looking for. Moved out the StateTransitionFilter to filter/core.py and added state_key field to the glue crawler model because AWS has a different key name for the states for different resources.